version: "3.7"
services:
    bot:
        image: hehelf/vk-bot:latest
        ports:
            - "${CALLBACK_PORT}:${CALLBACK_PORT}"
        container_name: vk-bot
        networks:
            - app-network
        volumes:
            - ./cert:/cert
        environment:
            - LOG_LEVEL
            - API_URL
            - VK_TOKEN
            - VK_GROUP
            - VK_GROUP_OWNER
            - CALLBACK_URL
            - CALLBACK_PORT
            - CALLBACK_TITLE
            - CALLBACK_SECRET
            - GH_TOKEN
            - GH_USER
            - GH_REPOSITORY
            - MIGRATE_PATH
            - PG_USER
            - PG_PASSWORD
            - PG_DB
            - PG_HOST
        depends_on:
            - pgdb
    pgdb:
        image: postgres:12.7-buster
        restart: unless-stopped
        ports:
            - "5432:5432"
        environment:
            - POSTGRES_USER=${PG_USER}
            - POSTGRES_PASSWORD=${PG_PASSWORD}
            - POSTGRES_DB=${PG_DB}
            - PGDATA=/var/lib/pgsql/data
        container_name: pgdb
        volumes:
            - ./pg-data:/var/lib/pgsql/data
        networks:
            - app-network
networks:
    app-network:
        driver: bridge
        driver_opts:
            com.docker.network.enable_ipv6: "false"
            com.docker.network.bridge.name: "app-network"